src/catchup/CatchupManagerImpl.cpp

```diff
@@ -63,24 +63,33 @@ CatchupManagerImpl::processLedger(LedgerCloseData const& ledgerData)
         mCatchupWork.reset();
     }
 
-    mSyncingLedgers.push_back(ledgerData);
+    auto addLedgerData = [&]() { mSyncingLedgers.push_back(ledgerData); };
 
-    auto contiguous =
-        lastReceivedLedgerSeq == mSyncingLedgers.back().getLedgerSeq();
     if (!mCatchupWork)
     {
-        uint32_t catchupTriggerLedger =
+        uint32_t checkpointLedger =
             mApp.getHistoryManager().nextCheckpointLedger(
-                lastReceivedLedgerSeq - 1) +
-            1;
+                lastReceivedLedgerSeq - 1);
 
+        // We wait until checkpoint + 1 ledger is closed before we start
```

Comment:
It's just that `checkpoint+2` is not really special, like you said nodes are publishing `checkpoint` when ledger `checkpoint+1` closes.
So waiting for `checkpoint+2` is really "wait for 5 seconds so that full validators will be done publishing" delay. I would prefer if instead we had an explicit delay (I imagine that we may have to tweak this number in the future).
I understand that his is more of a clarification, so we can punt that (so @sisuresh open an issue about it), as this is not really a problem today.

Reply:
Opened issue about using a delay - https://github.com/stellar/stellar-core/issues/2455